t/27-consoles-sshIucvconn.t

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/perl
+
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use Mojo::Base -strict, -signatures;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use consoles::sshIucvconn;
+use Test::MockObject;
+use Test::MockModule;
+use testapi;
+
+my $testapi_console = 'sshIucvconn';
+my $args = { hostname => 'testhost', password => 'testpass' };
+my $exec_flag = 1;
+$bmwqemu::vars{ZVM_GUEST} = 'guest.what';
+my $backend_mock = Test::MockObject->new();
+
+$backend_mock->mock('new_ssh_connection', sub($self, %args) {
+    my $ttyconn_mock = Test::MockObject->new;
+    $ttyconn_mock->mock('channel', sub {
+        my $channel_mock = Test::MockObject->new;
+        $channel_mock->mock('blocking', sub { 1 });
+        $channel_mock->mock('pty', sub { 1 });
+        $channel_mock->mock('exec', sub { $exec_flag });
+        return $channel_mock; });
+    $ttyconn_mock->mock('error', sub { undef });
+    return $ttyconn_mock;
+});
+
+$backend_mock->mock('start_ssh_serial', sub($self, %args) {
+    my $ssh_mock = Test::MockObject->new;
+    $ssh_mock->mock('blocking', sub { 1 });
+    $ssh_mock->mock('error', sub { 'unknown SSH error' });
+    my $serialchan_mock = Test::MockObject->new;
+    $serialchan_mock->mock('exec', sub { $exec_flag });
+    return ($ssh_mock, $serialchan_mock);
+});
+
+$backend_mock->mock('stop_ssh_serial', sub($self) { 1 });
+my $sshIucvconn_mock = Test::MockModule->new("consoles::$testapi_console");
+$sshIucvconn_mock->mock('backend', sub { $backend_mock });
+
+subtest 'connect_remote test' => sub {
+    my $c = consoles::sshIucvconn->new($testapi_console, $args);
+    ok($c->connect_remote($args), 'connect_remote executed successfully with mocked backend');
+    ok($c->kill_ssh(), 'kill_ssh executed successfully with mocked backend');
```

Comment:
1. Prefer no parentheses for easier reading of test statements as is done in most other test files as well
2. Mentioning the mocking is redundant as that can be expected in test modules, in particular the ones defining "unit tests"

```suggestion
    ok $c->connect_remote($args), 'connect_remote executed successfully';
    ok $c->kill_ssh(), 'kill_ssh executed successfully';
```